specification/storageactions/resource-manager/Microsoft.StorageActions/stable/2023-01-01/storageactions.json

```diff
@@ -556,12 +556,63 @@
         },
         "properties": {
           "x-ms-client-flatten": true,
-          "$ref": "#/definitions/StorageTaskProperties",
+          "$ref": "#/definitions/StorageTaskUpdateProperties",
           "description": "Properties of the storage task."
         }
       },
       "description": "Parameters of the storage task update request"
     },
+    "StorageTaskUpdateProperties": {
+      "type": "object",
+      "properties": {
+        "taskVersion": {
+          "readOnly": true,
+          "type": "integer",
+          "minimum": 1,
+          "maximum": 1,
+          "format": "int64",
+          "description": "Storage task version."
+        },
+        "enabled": {
+          "type": "boolean",
+          "description": "Storage Task is enabled when set to true and disabled when set to false"
+        },
+        "description": {
+          "type": "string",
+          "description": "Text that describes the purpose of the storage task"
+        },
+        "action": {
+          "$ref": "#/definitions/StorageTaskAction",
+          "description": "The storage task action that is executed"
+        },
+        "provisioningState": {
+          "readOnly": true,
```

Comment:
Per my understanding, "StorageTaskUpdateProperties" is only used in patch storage task input parameters. (It won't be used in output)
In this case, it looks the readonly properties are not necessary here, since user won't have chance to set them in update storage task.